docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
 # Developer Guide
 
+## Table of Contents
+* [Acknowledgements](#acknowledgements)
+* [Design and Implementation](#design--implementation)
+  * [Application Lifecycle](#application-lifecycle)
+  * [Parser Component](#parser-component)
+  * [Command Component](#command-components)
+  * [Storage Component](#storage-component)
+  * [UI Component](#ui-component)
+* [Product Scope](#product-scope)
+* [Non Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Instructions for Manual Testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. For the main structure of the program, we have adapted the project structure
+from the Individual Project of Man Juncheng at [Link](https://github.com/spinoandraptos/ip/tree/master) </li>
+2. For JUnit testing, we have adapted the testing codes from the AddressBook level-2
+codes at [Link](https://github.com/se-edu/addressbook-level2)</li>
+3. For the developer's guide, we have referenced the developer's guide from AddressBook
+level-3 [Link](https://se-education.org/addressbook-level3/DeveloperGuide.html)
+
+# Design & implementation
+
+## Overall Architecture
+The main execution of the QuizHub application will concern 4 components which
+are the QuizHub, Ui, Parser and Command packages.
+
+![](./UML/architecture.jpg)
```

Comment:
The architecture diagram is quite clear and concise, nice! The explanation for what each component does is really useful too